src/InteractiveSlide/InstantBook.php

```diff
@@ -147,97 +145,116 @@ function (CacheItemInterface $item) use ($configuration): mixed {
     }
 
     /**
-     * @throws \Throwable
+     * @throws InteractiveSlideException
+     * @throws InvalidArgumentException
      */
     private function getQuickBookOptions(Slide $slide, InteractionSlideRequest $interactionRequest): array
     {
         $resource = $interactionRequest->data['resource'] ?? null;
 
         if (null === $resource) {
-            throw new \Exception('Resource not set.');
+            throw new InteractiveSlideException('Resource not set.', 400);
         }
 
+        $start = (new \DateTime())->setTimezone(new \DateTimeZone('UTC'));
+        $startFormatted = $start->format('c');
+
         return $this->interactiveSlideCache->get(self::CACHE_KEY_OPTIONS_PREFIX.$resource,
-            function (CacheItemInterface $item) use ($slide, $resource, $interactionRequest) {
+            function (CacheItemInterface $item) use ($slide, $resource, $interactionRequest, $start, $startFormatted) {
                 $item->expiresAfter(new \DateInterval(self::CACHE_LIFETIME_QUICK_BOOK_OPTIONS));
 
-                /** @var User|ScreenUser $activeUser */
-                $activeUser = $this->security->getUser();
-                $tenant = $activeUser->getActiveTenant();
+                try {
+                    /** @var User|ScreenUser $activeUser */
+                    $activeUser = $this->security->getUser();
+                    $tenant = $activeUser->getActiveTenant();
 
-                $interactive = $this->interactiveService->getInteractiveSlide($tenant, $interactionRequest->implementationClass);
+                    $interactive = $this->interactiveService->getInteractiveSlide($tenant, $interactionRequest->implementationClass);
 
-                if (null === $interactive) {
-                    throw new \Exception('InteractiveSlide not found');
-                }
+                    if (null === $interactive) {
+                        throw new InteractiveSlideException('InteractiveSlide not found', 400);
+                    }
 
-                // Optional limiting of available resources.
-                $this->checkPermission($interactive, $resource);
+                    // Optional limiting of available resources.
```

Comment:
How is this "Optional"? If I understand the code a better comment would be

"Check if instant booking is allowed for the given resource"